java/tests/com/google/privacy/differentialprivacy/BoundedSumTest.java

```diff
@@ -678,4 +678,216 @@ private static void testForBias(
+            .thenAnswer(invocation -> new GaussianNoise().computeConfidenceInterval(
+                    (Double) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Double) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], (Double) invocation.getArguments()[4], (Double) invocation.getArguments()[5]));
+    sum =
+            BoundedSum.builder()
+                    .epsilon(0.5)
+                    .delta(0.9)
+                    .noise(noise)
+                    .maxPartitionsContributed(15)
+                    .lower(1)
+                    .upper(5)
+                    .build();
+    sum.addEntry(1);
+    sum.computeResult();
+
+    // The result interval = (-0.5760855815594823, 2.5760855815594823), but the bounds are negative,
+    // therefore it should be clamped to (0, 2.5760855815594823).
+    assertThat(sum.computeConfidenceInterval(0.152145599)).isEqualTo(ConfidenceInterval.create(0, 2.5760855815594823));
+  }
+
+  @Test
+  public void computeConfidenceInterval_forLaplaceNoise() {
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed
+    // as a value instead of null, in order to pass the checks.
+    when(noise.computeConfidenceInterval(anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Double) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Double) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], null, (Double) invocation.getArguments()[5]));
+    sum =
+            BoundedSum.builder()
+                    .epsilon(0.1)
+                    .noise(noise)
+                    .delta(0.5)
+                    .maxPartitionsContributed(1)
+                    .lower(1)
+                    .upper(10)
+                    .build();
+    sum.addEntry(10);
+    sum.computeResult();
+
+    assertThat(sum.computeConfidenceInterval(0.5)).isEqualTo(ConfidenceInterval.create(3.068528194400547, 16.931471805599454));
+  }
+
+  @Test
+  public void computeConfidenceInterval_whenComputeResultWasNotCalled_throwsException_forLong() {
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed
+    // as a value instead of null, in order to pass the checks.
+    when(noise.computeConfidenceInterval(anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], null, (Double) invocation.getArguments()[5]));
+    BoundedSum.builder()
```

Comment:
can't we use default sum?

Reply:
Default sum has -+Double.MAX_VALUE lower and upper bounds. For consistency with other tests i think it would be better to change the lower and upper bounds by building sum this way. Should i change it, WDYT?